vm/src/builtins/builtinfunc.rs

```diff
@@ -140,9 +139,26 @@ impl PyBuiltinFunction {
     }
 }
 
+// PyMethodDescrObject in
+// https://github.com/python/cpython/blob/main/Include/descrobject.h
+// and
+// https://github.com/python/cpython/blob/main/Objects/descrobject.c
 #[pyclass(module = false, name = "method_descriptor")]
 pub struct PyBuiltinMethod {
     value: PyNativeFuncDef,
+    dtype: Option<PyTypeRef>,
+}
+
+impl PyBuiltinMethod {
+    pub fn with_dtype(mut self, dtype: PyTypeRef) -> Self {
```

Comment:
`pub(crate)` not to expose to users.

But, if all methods finally have `d_type` before fully initialized, I suggest to not to create this method but adding the parameter to each creation methods not to let user failure.

Reply:
Yeap, totally! Now `mmtype` is just `PyTypeRef`